azurerm/resource_arm_key_vault.go

```diff
@@ -84,6 +84,34 @@ func resourceArmKeyVault() *schema.Resource {
 							Required:     true,
 							ValidateFunc: validateUUID,
 						},
+						"application_id": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateUUID,
+						},
+						"certificate_permissions": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+								ValidateFunc: validation.StringInSlice([]string{
+									string(keyvault.All),
+									string(keyvault.Create),
+									string(keyvault.Delete),
+									string(keyvault.Deleteissuers),
+									string(keyvault.Get),
+									string(keyvault.Getissuers),
+									string(keyvault.Import),
+									string(keyvault.List),
+									string(keyvault.Listissuers),
+									string(keyvault.Managecontacts),
+									string(keyvault.Manageissuers),
+									string(keyvault.Setissuers),
+									string(keyvault.Update),
```

Comment:
Wondering if there's an array/slice of all permissions, so that we don't need to update this list after Azure introduces new ones. 🤔

Reply:
it's exposed as a collection of Constants, so it might be hard in the short term - however we could ask the SDK team to expose that as a function?

Reply:
Up to you, it's no big deal - I just thought it may save us some pain/work in the future.
Feel free to 🚢  it as is.